exporters/geneva/src/unix_domain_socket_data_transport.cc

```diff
@@ -0,0 +1,73 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+#include "opentelemetry/exporters/geneva/metrics/unix_domain_socket_data_transport.h"
+#include "opentelemetry/exporters/geneva/metrics/macros.h"
+
+OPENTELEMETRY_BEGIN_NAMESPACE
+namespace exporter {
+namespace geneva {
+namespace metrics {
+
+UnixDomainSocketDataTransport::UnixDomainSocketDataTransport(
+    const std::string &connection_string) {
+  addr_.reset(new SocketTools::SocketAddr(connection_string.c_str(), true));
+}
+
+bool UnixDomainSocketDataTransport::Connect() noexcept {
+  if (!connected_) {
+    socket_ = SocketTools::Socket(socketparams_);
+    connected_ = socket_.connect(*addr_);
+    if (!connected_) {
+      LOG_ERROR("Geneva Exporter: UDS::Connect failed");
+      return false;
+    }
+  }
+  return true;
+}
+
+bool UnixDomainSocketDataTransport::Send(char const *data,
+                                         uint16_t length) noexcept {
+  int error_code = 0;
+  if (connected_) {
```

Comment:
Is lock needed to protect the send?

Reply:
This class is not thread safe intentionally, as the calling exporter will ensure that the metrics data is send sequentially.